src/main/java/pl/allegro/tech/opel/OpelParser.java

```diff
@@ -343,14 +347,36 @@ protected Rule fromStringLiteral(String string) {
         return String(string).label("'" + string + "'");
     }
 
+    /**
+     * Escapes special characters in string literals.
+     * Explicitly converts \t and \n to their corresponding whitespace characters.
+     * All other escaped characters are left as they are.
+     */
     protected String escapeString(String string) {
+        Matcher matcher = escapeCharacterPattern.matcher(string);
         StringBuilder result = new StringBuilder();
-        for (int i = 0; i < string.length(); i++) {
-            if (string.charAt(i) == '\\') {
-                i++;
+        while (matcher.find()) {
+            String replacement;
+            switch (matcher.group()) {
+                case "\\t":
+                    replacement = "\t";
+                    break;
+                case "\\r":
+                    replacement = "\r";
+                    break;
+                case "\\n":
+                    replacement = "\n";
+                    break;
+                case "\\\\":
+                    replacement = "\\\\";
+                    break;
+                default:
+                    replacement = matcher.group().substring(1);
```

Comment:
Won't this still strip `\`? So `\a` will become `a`. Do we want this to happen?
In that case `'apple'.startsWith('\a')` will return `true`, but should return `false`.

I think we should choose one consistent approach:
1. without escape sequences 
`\t`, `\n`, `\a`, `\x` are just literal strings, nothing is replaced with tabs, new lines and so on

2. with escape sequences 
a) `\t`, `\n` are replaced with tabs and new lines
b) `\a`, `\x` throws an error - illegal escape character

Otherwise we will have this unpredictable situation where
`'\' + variable` will return different value depending on the `variable`

Reply:
> Won't this still strip \? So \a will become a. Do we want this to happen?
In that case 'apple'.startsWith('\a') will return true, but should return false.

This is the current behaviour, changing this safely would require analysis of all expressions in the wild.

Reply:
we decided to not change any behaviour for now, just add tests for documentation purposes